nc/nc.go

```diff
@@ -287,30 +370,74 @@ func (c *Client) receiveMessage(_ mqtt.Client, msg mqtt.Message) {
 	}
 }
 
-func (c *Client) sendDeliveryConfirmation(messageID string) error {
-	payload, err := json.Marshal(ConfirmationPayload{
-		MessageID: messageID,
-	})
-	if err != nil {
-		return fmt.Errorf("marshaling confirmation payload: %w", err)
-	}
-	if token := c.client.Publish(topicDelivered, 1, false, payload); token.Wait() && token.Error() != nil {
-		return token.Error()
+func (c *Client) ncClientManagementLoop(ctx context.Context) error {
+	managementChan := make(chan interface{})
+
+	log.Println(logPrefix + " starting management loop")
+
+	var client mqtt.Client
+
+	credentialsInvalidated := false
+	credentials, err := c.credsFetcher.GetCredentialsFromConfig()
+	if err == ErrInvalidCredentials {
```

Comment:
If I understand correctly, `errors.Is` does a shallow compare, so it wouldn't work in this case. We'd need to compare to the error object returned by library, but the error we are looking for is not public(or at least not documented).

Reply:
it checks with `==` also inside the function the two members. but on top of this it also goes thru the errors chain.
Actually [errors.Is](https://pkg.go.dev/errors) is the recommended way in the documentation, instead of `==`.

Reply:
`==` will not give us anything because we are not relying on some error object but a particular string('not Authorized'). https://go.dev/play/p/KVqtKMnNIpq .
But it's all good actually, since the `mqtt` library defines and exports the error object, it's just not documented. Had to go to dig in the source code to find it. :smile: